From 072cd84c75d93c388fc857f2c91cd12324bbbb08 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Wed, 11 Jul 2012 08:05:05 +0200
Subject: [PATCH 21/47] pwm: fix used-uninitialized warning in pwm_get()

Some versions of GCC don't seem no notice that the initialization of the
index variable is tied to that of the chip variable and falsely report
it as potentially being used uninitialized. However, to save anybody
else from tripping over this, we now initialize the index variable
unconditionally.

Originally-by: Stephen Warren <swarren@nvidia.com>
Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 drivers/pwm/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pwm/core.c b/drivers/pwm/core.c
index fa4d79a..b4a282e 100644
--- a/drivers/pwm/core.c
+++ b/drivers/pwm/core.c
@@ -389,9 +389,9 @@ struct pwm_device *pwm_get(struct device *dev, const char *con_id)
 	struct pwm_device *pwm = ERR_PTR(-EAGAIN);
 	const char *dev_id = dev ? dev_name(dev) : NULL;
 	struct pwm_chip *chip = NULL;
+	unsigned int index = 0;
 	unsigned int best = 0;
 	struct pwm_lookup *p;
-	unsigned int index;
 	unsigned int match;
 
 	/*
-- 
1.7.9.5

